src/coreComponents/physicsSolvers/contact/SolidMechanicsAugmentedLagrangianContact.cpp

```diff
@@ -248,21 +260,46 @@ void SolidMechanicsAugmentedLagrangianContact::implicitStepSetup( real64 const &
                                         elemsToFaces,
                                         rotationMatrix );
 
+
+    // Set the tollerances
+    computeTolerances( domain );
+
+    // Set array to update penalty coefficients
+    arrayView2d< real64 > const dispJumpUpdPenalty =
+      subRegion.getReference< array2d< real64 > >( viewKeyStruct::dispJumpUpdPenaltyString() );
+
+    arrayView1d< real64 const > const normalDisplacementTolerance =
+      subRegion.getReference< array1d< real64 > >( viewKeyStruct::normalDisplacementToleranceString() );
+    arrayView1d< real64 > const & slidingTolerance =
+      subRegion.getReference< array1d< real64 > >( viewKeyStruct::slidingToleranceString() );
+    arrayView1d< real64 const > const & normalTractionTolerance =
+      subRegion.getReference< array1d< real64 > >( viewKeyStruct::normalTractionToleranceString() );
+
+    arrayView1d< real64 const > const area = 
+      subRegion.getElementArea().toViewConst();
+
     arrayView2d< real64 > const
     penalty = subRegion.getField< fields::contact::penalty >().toView();
 
     // Set the penalty coefficients
     // TODO: This is only temporary. The setting of penalty will be adaptive in the final version.
     forAll< parallelDevicePolicy<> >( subRegion.size(), [=] GEOS_HOST_DEVICE ( localIndex const k )
     {
-      penalty[k] [0] = 1.e+7;
-      penalty[k] [1] = 1.e+7;
+
+      penalty[k] [0] = normalTractionTolerance[k]/(normalDisplacementTolerance[k]*area[k] );
+      penalty[k] [1] = penalty[k][0] * 1.0e-01;
```

Comment:
why are there 2 penalties?

Reply:
The first component updates the normal component of the Lagrange multiplier, while the second component updates the tangential component.